build.proj

```diff
@@ -122,6 +122,20 @@
     <WriteLinesToFile File="$(CompletedSemaphorePath)ReportPoisonUsage.complete" Overwrite="true" />
   </Target>
 
+  <Target Name="WriteVersionsToFile"
+          AfterTargets="Build">
+
+    <WriteLinesToFile Lines="$(SdkProductVersion)" File="$(BaseOutputPath)\build-info\SdkVersion.txt" Overwrite="True" />
+    <WriteLinesToFile Lines="$(AspNetCoreProductVersion)" File="$(BaseOutputPath)\build-info\AspNetCoreVersion.txt" Overwrite="True" />
+    <WriteLinesToFile Lines="$(RuntimeProductVersion)" File="$(BaseOutputPath)\build-info\RuntimeVersion.txt" Overwrite="True" />
```

Comment:
`SdkProductVersion` and friends are defined in `eng/Versions.props`. Is it safe to assume they will always be correct?

Reply:
I don't think we want to depend on that, especially since this will partially be used for catching human error cases.  Ideally I think we get this information directly from the built SDK, either package or tarball versions or actually running and parsing `dotnet --info` or similar.

Reply:
Updated now. What do you think?

I also added a check for https://github.com/dotnet/source-build/issues/2379.